### PR TITLE
 Fix UX issues with bug report dialog

### PR DESCRIPTION
**Pull request by dbkr**
* Make it use BaseDialog / DialogButtons (also gives it has a top-right 'x' &
   escape to cancel works)
 * Stop misusing the 'danger' CSS class on the buttons. There is nothing dangerous
   about submitting logs.
 * Continued campaign against 'Click here' links.

Fixes vector-im/riot-web#6622

Based off https://github.com/matrix-org/matrix-react-sdk/pull/1860 because I keep needing to add things to DialogButtons and it would conflict.

### CODE REVIEW

File: res/css/_common.scss [hunk 1]
```diff
@@ -250,6 +250,7 @@ textarea {
 .mx_Dialog button.danger, .mx_Dialog input[type="submit"].danger {
     background-color: $warning-color;
     border: solid 1px $warning-color;
+    color: $accent-fg-color;
```
Comment (lukebarnard1) [inline res/css/_common.scss]: From a code perspective, fine. From a UI perspective though, this text should very much be white.

File: src/components/views/dialogs/BaseDialog.js [hunk 2]
```diff
@@ -41,6 +42,13 @@ export default React.createClass({
         // cancelled (BaseDialog itself only calls this with false).
         onFinished: PropTypes.func.isRequired,
 
+        // Whether the dialog should have a 'close' button that will
+        // cause the dialog to be cancelled. This should only be set
+        // to true if there is nothing the app can sensibly do if the
```
Comment (lukebarnard1) [inline src/components/views/dialogs/BaseDialog.js]: Should this not be `false`?

Comment (lukebarnard1) [reply src/components/views/dialogs/BaseDialog.js]: Worth mentioning the default.

File: src/components/views/dialogs/SessionRestoreErrorDialog.js [hunk 3]
```diff
@@ -31,63 +31,73 @@ export default React.createClass({
...
+                </button>
+            </DialogButtons>;
+        } else {
+            dialogButtons = <DialogButtons primaryButton={_t("Refresh")}
+                onPrimaryButtonClick={this._onRefreshClick}
+                focus={true}
+                hasCancel={false}
+            >
+                <button onClick={this._onClearStorageClick} className="danger">
+                    { _t("Clear Storage and Sign Out") }
```
Comment (lukebarnard1) [inline src/components/views/dialogs/SessionRestoreErrorDialog.js]: This looks identical to the `if` block apart from the text "Refresh". 

OH. I see what this is doing now. Maybe factor out the "Clear Storage..." button and refer to it twice.

File: src/components/views/dialogs/SessionRestoreErrorDialog.js [hunk 4]
```diff
@@ -31,63 +31,73 @@ export default React.createClass({
...
-                    "history will be unreadable.") }</p>
+                    <p>{ _t("We encountered an error trying to restore your previous session.") }</p>
 
                     <p>{ _t("If you have previously used a more recent version of Riot, your session " +
                     "may be incompatible with this version. Close this window and return " +
                     "to the more recent version.") }</p>
 
-                    { bugreport }
+                    <p>{ _t("Clearing your browser's storage may fix the problem, but will sign you " +
+                    "out and cause any encrypted chat history to become unreadable.") }</p>
```
Comment (lukebarnard1) [inline src/components/views/dialogs/SessionRestoreErrorDialog.js]: Indentation could be better here

File: res/css/_common.scss (hunk 1 above)
Comment (dbkr) [reply res/css/_common.scss]: Currently the text on all the red buttons is black in the dark theme

File: src/components/views/dialogs/BaseDialog.js (hunk 2 above)
Comment (dbkr) [reply src/components/views/dialogs/BaseDialog.js]: ah, good spot

File: src/components/views/dialogs/SessionRestoreErrorDialog.js (hunk 3 above)
Comment (dbkr) [reply src/components/views/dialogs/SessionRestoreErrorDialog.js]: mm, that is probably better. done

File: src/components/views/dialogs/SessionRestoreErrorDialog.js (hunk 4 above)
Comment (dbkr) [reply src/components/views/dialogs/SessionRestoreErrorDialog.js]: done

File: res/css/_common.scss (hunk 1 above)
Comment (lukebarnard1) [reply res/css/_common.scss]: Oh. But isn't accent-fg-color green?

Comment (lukebarnard1) [reply res/css/_common.scss]: Oh. Sorry, I should've looked at what the colours actually are.

**Pull request merged**